moon/controller_hauler_round2.py

```diff
@@ -125,21 +159,33 @@ def run(self):
                         print(self.sim_time, self.robot_name, "Bumper")
                         self.inException = True
                         self.go_straight(-1) # go 1m in opposite direction
-                        self.drive_around_rock(6) # assume 6m the most needed
+                        self.drive_around_rock(math.copysign(4, 1 if getrandbits(1) == 0 else -1)) # assume 6m the most needed
                         self.inException = False
                         continue
+
                 try:
+                    if self.excavator_waiting:
+                        self.turn(math.radians(360 if self.rover_angle > 0 else -360))
```

Comment:
why do you turn whole 360deg? Or is it "up to 360deg"? (otherwise I do understand the motivation for direction choice)

Reply:
This is a scenario where hauler lost visual on excavator. Typically this happens when excavator turned quicker than hauler or when hauler was trying to avoid an obstacle while following and lost contact then. In that case, hauler asks excavator to stop and hauler starts turning in the direction where excavator was seen last. Once it is spotted, the turn is interrupted and following routine resumes.

This whole following would ideally be a little more robust, excavator could be hiding behind a boulder and not be visible even after a 360 turn. So at least adding a random walk. And/or follow more robustly as not to lose excavator in first place.